src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/ModuleSpeed.kt

```diff
@@ -43,10 +49,36 @@ object ModuleSpeed : Module("Speed", Category.MOVEMENT) {
                 wait(1)
                 player.downwards(-1f)
             }
-
         }
+    }
+
+    private object HypixelHop : Choice("HypixelHop", modes) {
+        val playerNetworkMovementTick = handler<PlayerNetworkMovementTickEvent> { event ->
 
+            if (event.state != EventState.PRE)
+                return@handler
+
+            if (player.moving) {
+                if (player.isOnGround) {
+                    player.updatePosition(player.x, player.y + 9.1314E-4, player.z)
+                    player.jump()
+                    player.strafe(speed = ((if (getSpeed() < 0.56f) getSpeed() * 1.045f else 0.56f) * (1f + 0.13f * getSpeedEffectAmplifier())).toDouble())
+                } else if (player.velocity.y < 0.2)
+                    player.velocity.y -= 0.02
+                player.strafe(speed = (getSpeed() * 1.01889f).toDouble())
+            } else {
+                player.velocity.x = 0.0
+                player.velocity.z = 0.0
+            }
+        }
     }
 
-}
+    fun getSpeedEffectAmplifier() =
+        player.getStatusEffect(StatusEffects.SPEED)!!.amplifier.plus(1)
 
+    fun getSpeed(): Float {
```

Comment:
We already does have a speed function. Using mc.player.speed (not sure)

Reply:
It kinda differs I think.